File: Java/Graphs/ProblemsOnDfsBfs/733-flood-fill/flood-fill.java
class Solution {
    private static void dfs(int row,int col, int ans[][], int initialColor, int newColor, int delRow[], int delCol[]){
        ans[row][col] = newColor;
        int n= ans.length;
        int m = ans[0].length;

        for(int i=0;i<4;i++){
            int nrow = row + delRow[i];
            int ncol = col + delCol[i];
            if(nrow>=0 && nrow<n && ncol>=0 && ncol < m && ans[nrow][ncol] == initialColor && ans[nrow][ncol] != newColor){
                dfs(nrow,ncol,ans,initialColor,newColor,delRow,delCol);
            }
        }
    }
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        int ans[][] = image;
        int initialColor = image[sr][sc];
        int delRow[] = {-1,0,1,0};
        int delCol[] = {0,1,0,-1};
        dfs(sr,sc,ans,initialColor,color,delRow,delCol);
        return ans;
    }
}

File: Python/Graphs/ProblemsOnBFSAndDFS/733-flood-fill/flood-fill.py
class Solution:
    def dfs(self, row : int, col: int ,initial_color:int, new_color: int, ans: List[List[int]], delRow:List[int], delCol: List[int]):
        ans[row][col] = new_color
        n = len(ans)
        m = len(ans[0])

        for i in range(4):
            n_row = row + delRow[i]
            n_col = col + delCol[i]

            if(n_row >= 0 and n_row < n and n_col >=0 and n_col <m):
                if(ans[n_row][n_col] == initial_color and ans[n_row][n_col] != new_color):
                    self.dfs(n_row,n_col, initial_color, new_color, ans,delRow, delCol)
        
    def floodFill(self, image: List[List[int]], sr: int, sc: int, color: int) -> List[List[int]]:
        initialColor = image[sr][sc]
        ans = image
        delRow = [-1,0,1,0]
        delCol = [0,1,0,-1]
        self.dfs(sr,sc,initialColor, color,ans,delRow,delCol)
        return ans

